### DISCUSSION TITLE
Using Vanna for data transforms

### DISCUSSION THREAD

**Discussion by alexandreczg** [vanna-ai/vanna#241] [General]
Hello,

Firstly, I must appreciate this incredible project. It got me contemplating potential use cases within my organization. 

Currently, Vanna excels at generating queries from natural language - which is most impressive! However, I'd like to open up a discussion around expanding its capabilities to handle data transformations.

Let's consider a scenario where both the source and target schemas are known to the model. Additionally, our analysts would be providing natural language instructions to map source-to-target fields. Wouldn't it be fascinating if Vanna could generate, for instance, a DBT transform to manage this procedure?

In my opinion, such an enhancement seems quite plausible given Vanna's current capabilities. Nonetheless, I would greatly value input from the creators of this framework!

Best regards,
Alex

**Comment by zainhoda** [comment]
Thank you. This is a great idea. I'd like to see a proof of concept of this before we consider a broader implementation. It sounds like something that should be theoretically be possible but before we try it, I'm not sure what issues we'll run into.

If you're up for customization, this is how you can add your own additional LLM-based function to add this as a proof-of-concept if you're so inclined:
https://vanna.ai/docs/behavior-customization/#adding-an-additional-llm-based-function
